### PR TITLE
blas/gonum: add Zhpr

### PR DESCRIPTION
**Pull request by vladimir-ch**

### CODE REVIEW

File: blas/gonum/level2cmplx128.go [hunk 1]
```diff
@@ -549,6 +549,122 @@ func (Implementation) Zher2(uplo blas.Uplo, n int, alpha complex128, x []complex
...
+			if xi != 0 {
+				tmp := complex(alpha, 0) * xi
+				a := ap[ii : ii+i]
+				for j, v := range x[:i] {
+					a[j] += tmp * cmplx.Conj(v)
+				}
+
+				aii := real(ap[ii+i]) + alpha*real(cmplx.Conj(xi)*xi)
+				ap[ii+i] = complex(aii, 0)
+			} else {
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: ?

File: blas/gonum/level2cmplx128.go [hunk 2]
```diff
@@ -549,6 +549,122 @@ func (Implementation) Zher2(uplo blas.Uplo, n int, alpha complex128, x []complex
...
+
+	// Set up start index in X.
+	var kx int
+	if incX < 0 {
+		kx = (1 - n) * incX
+	}
+
+	// The elements of A are accessed sequentially with one pass through ap.
+
+	var ii int
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: Maybe `kk` to avoid confusion?

File: blas/testblas/common.go [hunk 3]
```diff
@@ -349,3 +349,62 @@ func max(a, b int) int {
...
 }
+
+// packHermitian returns an n×n Hermitian matrix in packed format stored in
+// the uplo triangle of the general matrix A.
+func packHermitian(uplo blas.Uplo, n int, a []complex128, lda int) []complex128 {
+	if n == 0 {
+		return nil
+	}
+	ap := make([]complex128, n*(n+1)/2)
+	var ii int
```
Comment (kortschak) [inline blas/testblas/common.go]: `kk`?

File: blas/testblas/common.go [hunk 4]
```diff
@@ -349,3 +349,62 @@ func max(a, b int) int {
...
+
+// unpackHermitian returns an n×n general Hermitian matrix (with stride n)
+// whose packed uplo triangle is stored on entry in ap.
+func unpackHermitian(uplo blas.Uplo, n int, ap []complex128) []complex128 {
+	if n == 0 {
+		return nil
+	}
+	a := make([]complex128, n*n)
+	lda := n
+	var ii int
```
Comment (kortschak) [inline blas/testblas/common.go]: And here?

File: blas/testblas/common.go [hunk 5]
```diff
@@ -349,3 +349,62 @@ func max(a, b int) int {
 	}
 	return a
 }
+
+// packHermitian returns an n×n Hermitian matrix in packed format stored in
+// the uplo triangle of the general matrix A.
+func packHermitian(uplo blas.Uplo, n int, a []complex128, lda int) []complex128 {
```
Comment (kortschak) [inline blas/testblas/common.go]: Can we have a round trip test for this pair of helpers?

File: blas/gonum/level2cmplx128.go (hunk 2 above)
Comment (vladimir-ch) [reply blas/gonum/level2cmplx128.go]: Done.

File: blas/testblas/common.go (hunk 5 above)
Comment (vladimir-ch) [reply blas/testblas/common.go]: Done.

File: blas/testblas/common.go (hunk 4 above)
Comment (vladimir-ch) [reply blas/testblas/common.go]: Done.

**Comment by vladimir-ch** [comment]
PTAL

File: blas/testblas/common_test.go [hunk 6]
```diff
@@ -2,8 +2,10 @@ package testblas
 
 import (
 	"math"
+	"math/cmplx"
 	"testing"
 
+	"golang.org/x/exp/rand"
```
Comment (kortschak) [inline blas/testblas/common_test.go]: This should be in its own import stanza between std and us.

**Comment by kortschak** [comment]
Please remember to send a PR adding the tests here and running check-generate.sh in netlib after this is merged.

File: blas/testblas/common_test.go (hunk 6 above)
Comment (vladimir-ch) [reply blas/testblas/common_test.go]: Done.

**Pull request merged**